### DISCUSSION TITLE
Welcome to gopro-plus Discussions!

### DISCUSSION THREAD

**Discussion by dustin** [dustin/gopro-plus#1] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by kikothemaster** [comment]
This should be written in java/kotlin/js.

**Reply by dustin** [reply]
I've spent a lot of days reverse engineering all of the messages and flows for all of the interactions with the GoPro backend service and building a set of commandline and web-based tools that let me interact with the service every way I needed to.  I also have been documenting all of the flows in OAS 3.

It's a bit rude to say I also should've written the tools in worse languages.  It was already a hard project, why should I make it harder for myself?

**Reply by kikothemaster** [reply]
Sorry, I didn't mean that in any rude way, I should've used word "could" instead of "should".

**Reply by dustin** [reply]
NP.  I started a documentation project so people can access GoPro Plus from whatever tools they wish to use.  I personally use Haskell for just about everything because it's the easiest language to write good software in.  This particular software project is a bit messy as it's in the state of a lot of experimentation, but making such broad changes in an experimental codebase like this is just so much easier.

The project isn't quite done, but Swagger will generate a ton of client stubs:  https://app.swaggerhub.com/apis/dlsspy/gopro-plus/1.0.0-oas3#/
